### DISCUSSION TITLE
Some questions moving from Nextcloud Phonetrack

### DISCUSSION THREAD

**Discussion by kaibagley** [Freika/dawarich#161] [Q&A]
Hi everyone,

First I want to say that I'm very excited to see this project grow, I wish I could contribute more than just bug reports but I know nothing about Ruby :(. I've been looking for something like this for ages

I currently use Nextcloud Phonetrack and GpxPod, and would love to move to something like DaWarIch which is much more interesting to look at, however there are some things that I'd like to know/ask:

1. I read https://github.com/Freika/dawarich/discussions/66 which is exciting, I'd love to be able to upload data to this all day every day but I'm not currently using a VPN to access my home server and would rather not have to set it up. In the mean time, I will host this on my main machine and periodically manually import my phonetrack exports into it for testing
2. I'm assuming that it's possible, but will other apps such as my beloved [ulogger](https://github.com/bfabiszewski/ulogger-android) be able to upload to the server?
3. I often go for runs, are there plans for "outdoor activity" tracking features such as colouring the track based on speed, heart rate, etc., or plots of historical summaries of this data? GpxPod is what I'm using now for this and it's not very feature rich. If not that is absolutely fine, I understand that this one is a lot to ask! Please let me know if you'd like GPX files of this type, I'd be happy to provide them.

Thanks so much for your time!

**Comment by Freika** [comment]
Hi!

1. Oauth is still not a priority, although I think it will be implemented at some point
2. I couldn't quickly find a data structure that is being send by ulogger to the server, can you provide me with an example? Or maybe ulogger can send a manually defined data structure?
3. Dawarich is a geolocation history visualization tool in the first place. This means, it might be possible to implement described features, but they are not a priority at the moment. Dawarich already supports import of GPX files, but I take it, your files from GpxPod weren't successfully imported?

**Reply by kaibagley** [reply]
Thanks for the response!

I don't know much about the ulogger data structure, but the developer says here https://github.com/bfabiszewski/ulogger-server/issues/188#issuecomment-1120154908 that POST parameters can be found in the mentioned region of code, so imagine the android app does something along those lines? I also found this commit on the original Phonetrack repo https://gitlab.com/eneiluj/phonetrack-oc/-/commit/545817b8d5c9255f3920f452234d0a5609310714 where they implemented ulogger functionality. I'm not fully attached to ulogger though, I just love its minimal battery drain.

My GPX files all imported perfectly fine except for a few files which seemed corrupted in some way.
